app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -146,18 +151,32 @@ public class MainActivity : AppCompatActivity(), ViewController, Router.Callback
         LocationServices.FusedLocationApi?.requestLocationUpdates(locationRequest) {
             location: Location -> currentLocation = location
             val routeModeView = findViewById(R.id.route_mode) as RouteModeView
-            routeModeView.onLocationChanged(location)
+            if (routeModeView != null && routeModeView.getVisibility() == View.VISIBLE) {
```

Comment:
maybe could have a check in the presenter since it keeps track of routing mode being enabled?